File: test09/distinct_lines.py
#!/usr/bin/env python3

import sys, re

maxd = int(sys.argv[1])
seen = {}
nread = 0

for line in sys.stdin:
   nread += 1
   line = re.sub('\s+', '', line.lower())
   seen[line] = 1
   ndist = len(seen.keys())
   if (ndist >= maxd):
      print("%d distinct lines seen after %d lines read." % (maxd, nread))
      sys.exit()
      
print("End of input reached after %d lines read - %2d different lines not seen." % (nread, maxd))

File: test07/distinct_lines.pl
#!/usr/bin/perl -w

while ($line = <STDIN>) {
   $key = lc $line;
   $key =~ s/\s+//g;
   
   if (!$seen{$key}) {
      $seen{$key} = 1;
   }
   $nlines++;
   
   if (keys %seen >= $ARGV[0]) {
      print "$ARGV[0] distinct lines seen after $nlines lines read.\n";
      exit;
   }
}

print "End of input reached after $nlines lines read - $ARGV[0] different lines not seen.\n";
